core-rust/mesh-api-server/src/mesh_api/handlers/account_balance.rs

```diff
@@ -15,7 +15,7 @@ pub(crate) async fn handle_account_balance(
     )
     .map_err(|err| err.into_response_error("account_identifier"))?;
 
-    let database = state.state_manager.database.snapshot();
+    let database = state.state_manager.database.access_direct();
```

Comment:
Why have we changed to use `access_direct` here rather than `snapshot`? If we don't use `scoped_at`, then we can observe a non-snapshot read, which probably breaks atomicity of response.

e.g. I could trade `A` for `B` and if I do a well-timed read at the same time, I could theoretically observe both `A` and `B` (or neither `A` and `B`) in my account.

Reply:
Decided to revert to `snapshot()` as per [this conversation](https://rdxworks.slack.com/archives/C071C6DB8TG/p1733841098517859?thread_ts=1732877188.486819&cid=C071C6DB8TG)